### PR TITLE
Add mountnode datepicker compat

### LINKED ISSUE #28625: [Feature]: Add `mountNode` property to Combobox
### Library

React Northstar / v0 (@fluentui/react-northstar)

### Describe the feature that you would like added

The Portal component's `mountNode` property lets the user specify the target DOM node into which the Portal/popover/flyout will render. This is a very useful property for cases where `<body>` or inline won't do, such as the PowerApps Canvas surface. 

The `InfoButton` component already has this available; it's exposed via the larger [`popover`](https://github.com/microsoft/fluentui/blob/99e835ade7328fcb47114f5d5d293c36c2665ae1/packages/react-components/react-infobutton/src/components/InfoButton/InfoButton.types.ts#L10) property. Something like that would be agreeable for Combobox, but I'll leave the implementation details up to the experts. 

### Have you discussed this feature with our team

GeoffCoxMSFT

### Additional context

This is an internal feature request from the PowerApps App fabric team.

### Validations

- [X] Check that there isn't already an issue that request the same feature to avoid creating a duplicate.

### PR DESCRIPTION
**Pull request by GeoffCoxMSFT**
## Summary
Allows callers to change the mountNode of the portal used to display the dropdown part for DatePicker compat.

## Changes
- Added mountNode property to props and state
- Updated use hook to set state from props
- Updated render to pass mountNode to Portal

## Issues
Fixes #28625

### CODE REVIEW

**Comment by fabricteam** [comment]
## 📊 Bundle size report

| Package & Exports                                                                                                    |      Baseline (minified/GZIP) |                            PR |                                                                                                                                                                                                 Change |
| :------------------------------------------------------------------------------------------------------------------- | ----------------------------: | ----------------------------: | -----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------: |
| <samp>react-datepicker-compat</samp> <br /> <abbr title='bundle-size/DatePicker.fixture.js'>DatePicker Compat</abbr> | `218.493 kB`<br />`58.297 kB` | `218.541 kB`<br />`58.318 kB` | `48 B` <br />`21 B`  |

<sub>🤖 This report was generated against <a href='https://github.com/microsoft/fluentui/commit/253798f25326e92ba19b313c47a9c9530af12526'>253798f25326e92ba19b313c47a9c9530af12526</a></sub>
<div id="bundleSizeComment9423"/>

**Comment by fabricteam** [comment]
## 🕵 fluentuiv9 No visual regressions between this PR and main
<div id="vrtCommentfluent-ui_VRT_Pipeline_v9"/>

File: change/@fluentui-react-datepicker-compat-d51eda0f-25b4-4190-b43a-79e3714272da.json [hunk 1]
```diff
@@ -0,0 +1,7 @@
+{
+  "type": "minor",
+  "comment": "Added mountNode to DatePicker compat",
```
Comment (khmakoto) [inline change/@fluentui-react-datepicker-compat-d51eda0f-25b4-4190-b43a-79e3714272da.json:3]:
```suggestion
  "comment": "feat: Added mountNode to DatePicker compat.",
```

File: packages/react-components/react-datepicker-compat/src/components/DatePicker/DatePicker.types.ts [hunk 2]
```diff
@@ -3,6 +3,7 @@ import { Input } from '@fluentui/react-input';
 import type { ComponentProps, ComponentState, Slot } from '@fluentui/react-utilities';
 import type { CalendarProps } from '../Calendar/Calendar.types';
 import type { CalendarStrings, DateFormatting } from '../../utils';
+import { PortalProps } from '@fluentui/react-portal';
```
Comment (khmakoto) [inline packages/react-components/react-datepicker-compat/src/components/DatePicker/DatePicker.types.ts:6]:
```suggestion
import type { PortalProps } from '@fluentui/react-portal';
```

File: packages/react-components/react-datepicker-compat/src/components/DatePicker/DatePicker.types.ts [hunk 3]
```diff
@@ -11,215 +12,217 @@ export type DatePickerSlots = {
...
 
 /**
  * State used in rendering DatePicker.
  */
-export type DatePickerState = ComponentState<DatePickerSlots> & {
-  disabled: boolean;
-  inlinePopup: boolean;
-};
+export type DatePickerState = ComponentState<DatePickerSlots> &
+  Pick<PortalProps, 'mountNode'> & {
```
Comment (khmakoto) [inline packages/react-components/react-datepicker-compat/src/components/DatePicker/DatePicker.types.ts:222]: Shouldn't this be?

```suggestion
  Pick<PortalState, 'mountNode'> & {
```

File: packages/react-components/react-datepicker-compat/src/components/DatePicker/DatePicker.types.ts [hunk 4]
```diff
@@ -3,6 +3,7 @@ import { Input } from '@fluentui/react-input';
 import type { ComponentProps, ComponentState, Slot } from '@fluentui/react-utilities';
 import type { CalendarProps } from '../Calendar/Calendar.types';
 import type { CalendarStrings, DateFormatting } from '../../utils';
+import type { PortalProps } from '@fluentui/react-portal';
```
Comment (khmakoto) [inline packages/react-components/react-datepicker-compat/src/components/DatePicker/DatePicker.types.ts:6]: You probably need to import `PortalState` now as well

```suggestion
import type { PortalProps, PortalState } from '@fluentui/react-portal';
```

**Review by khmakoto — APPROVED** [review]

**Pull request merged**